src/app.js

```diff
@@ -3,29 +3,183 @@ go.app = function() {
     var App = vumigo.App;
     var Choice = vumigo.states.Choice;
     var ChoiceState = vumigo.states.ChoiceState;
+    var FreeText = vumigo.states.FreeText;
     var EndState = vumigo.states.EndState;
 
     var GoApp = App.extend(function(self) {
-        App.call(self, 'states:start');
+        App.call(self, 'state_welcome');
+        var $ = self.$;
+        // var interrupt = true;
 
-        self.states.add('states:start', function(name) {
-            return new ChoiceState(name, {
-                question: 'Hi there! What do you want to do?',
+        self.states.add('state_welcome', function(name) {
+            var question = $("Welcome! To report a malaria case, please enter your facility code. For example, 543456");
+            var error = $("The facility code is invalid. Please enter again.");
+            return new FreeText(name, {
+                  question: question,
+                  check: function(content) {
+                      if (go.utils.is_valid_name(content) || true ) {
```

Comment:
yeah... it's very much WIP.... I did that to make the test pass for now.

not ready for review by any stretch :1234: